plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -153,9 +154,49 @@ public void testCreateOrReplaceTableAsSelect()
                         .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
                         .build());
 
+        assertFileSystemAccesses(
+                "CREATE OR REPLACE TABLE test_create_or_replace_as_select AS SELECT 2 different_col_name",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", "InputFile.exists"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000000.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000001.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.exists"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(LAST_CHECKPOINT, "_last_checkpoint", "InputFile.newStream"))
+                        .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
+                        .build());
+
         assertUpdate("DROP TABLE test_create_or_replace_as_select");
     }
 
+    @Test
+    public void testCreateOrReplaceTableWithCheckpoints()
```

Comment:
Probably `testCheckpointing()` from the smoke test is enough.

I'm not sure whether this test adds any value :( - sorry for misleading you.